### PR TITLE
Update docker-compose environment variables and README

### PR DESCRIPTION
**Pull request by dianabarsan**
Adds missing environment variables to docker-compose and to README.

### CODE REVIEW

**Comment by mrjones-plip** [comment]
Oh yeah! I went to smoke test this branch and I'm unable to get this to work.  I get the error `volume name is too short`:

```
➜  cht-upgrade-service git:(update-env) ls /home/mrjones/Documents/MedicMobile/4.0.0-alpha-compose/srv
_dbs.couch  _nodes.couch  _replicator.couch  shards  _users.couch
➜  cht-upgrade-service git:(update-env) ls /home/mrjones/Documents/MedicMobile/4.0.0-alpha-compose/4-core.yml
/home/mrjones/Documents/MedicMobile/4.0.0-alpha-compose/4-core.yml
➜  cht-upgrade-service git:(update-env) ls 
docker-compose.yml  Dockerfile  package.json  package-lock.json  README.md  src  test
➜  cht-upgrade-service git:(update-env) COUCHDB_SECRET=foo-bar-bash-smang COUCHDB_DATA=/home/mrjones/Documents/MedicMobile/4.0.0-alpha-compose/srv CHT_COMPOSE_PATH=/home/mrjones/Documents/MedicMobile/4.0.0-alpha-compose/4-core.yml COUCHDB_USER=medic COUCHDB_PASSWORD=password  docker-compose up
WARNING: The DOCKER_CONFIG_PATH variable is not set. Defaulting to a blank string.
Creating cht-upgrade-service_cht-upgrade-service_1 ... error

ERROR: for cht-upgrade-service_cht-upgrade-service_1  Cannot create container for service cht-upgrade-service: create .: volume name is too short, names should be at least two alphanumeric characters

ERROR: for cht-upgrade-service  Cannot create container for service cht-upgrade-service: create .: volume name is too short, names should be at least two alphanumeric characters
ERROR: Encountered errors while bringing up the project.
```

It's also not clear to me if CouchDB should be up and running before trying to run this. I tried with and without it running. 

Happy to file this in a separate ticket if it stands on it's own and is unrelated to this branch!

**Comment by dianabarsan** [comment]
CouchDB should _not_ be running. 
CHT_COMPOSE_PATH should be the folder that includes both yml that start couch and CHT.

**Comment by mrjones-plip** [comment]
> CouchDB should _not_ be running. CHT_COMPOSE_PATH should be the folder that includes both yml that start couch and CHT.

Hmm - maybe my funny compose file names are messing things up? (`4-core.yml` and `4-couch.yml`)

I'll try with the original ones as they were when uploaded by CI to staging

**Comment by dianabarsan** [comment]
Did you make any changes to the `cht-upgrade-service` docker compose file?

**Comment by dianabarsan** [comment]
I think you're missing `DOCKER_CONFIG_PATH`. That was added to allow downloading images from the password protected Amazon ECR - it means you'd share the docker logins from the host to the docker process in the container. 
Does anything change if you set that environment variable?

**Comment by mrjones-plip** [comment]
> Did you make any changes to the `cht-upgrade-service` docker compose file?

Nope!   sanity checking...mine is MD5 hash of `776b0a2d4f9f72ac4a654b63fe4fe8c7` which matches:

```
➜  cht-upgrade-service git:(update-env) curl -o upto-date-compose.yml https://raw.githubusercontent.com/medic/cht-upgrade-service/update-env/docker-compose.yml
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   958  100   958    0     0   3878      0 --:--:-- --:--:-- --:--:--  3878

➜  cht-upgrade-service git:(update-env) ✗ md5sum upto-date-compose.yml 
776b0a2d4f9f72ac4a654b63fe4fe8c7  upto-date-compose.yml

➜  cht-upgrade-service git:(update-env) md5sum docker-compose.yml 
776b0a2d4f9f72ac4a654b63fe4fe8c7  docker-compose.yml

➜  cht-upgrade-service git:(update-env) COUCHDB_SECRET=foo-bar-bash-smang COUCHDB_DATA=/home/mrjones/Documents/MedicMobile/4.0.0-alpha-compose/srv CHT_COMPOSE_PATH=/home/mrjones/Documents/MedicMobile/4.0.0-alpha-compose COUCHDB_USER=medic COUCHDB_PASSWORD=password  docker-compose up
WARNING: The DOCKER_CONFIG_PATH variable is not set. Defaulting to a blank string.
Creating cht-upgrade-service_cht-upgrade-service_1 ... error

ERROR: for cht-upgrade-service_cht-upgrade-service_1  Cannot create container for service cht-upgrade-service: create .: volume name is too short, names should be at least two alphanumeric characters

ERROR: for cht-upgrade-service  Cannot create container for service cht-upgrade-service: create .: volume name is too short, names should be at least two alphanumeric characters
ERROR: Encountered errors while bringing up the project.
```

**Comment by mrjones-plip** [comment]
ah - the issue is `DOCKER_CONFIG_PATH`. we set it as a volume path here:

```yml
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ${DOCKER_CONFIG_PATH}:/config
```

So without specifying it, the compose call fails with the `volume name is too short` error.  I set it to the same path as where my compose file is and the container was able to start.

so I think either set a default in the compose file or change it to be required?

**Comment by dianabarsan** [comment]
I was reluctant to set a default because maybe people will want to use it in different operating systems (?). Maybe just making it required would work. Open to suggestions.

**Comment by mrjones-plip** [comment]
I still don't quite know what `DOCKER_CONFIG_PATH` is used for. Is this for the [daemon config](https://docs.docker.com/config/daemon/#configure-the-docker-daemon) or the credentials to the AWS ECR config or something else?

What little I know this, I would say make it a sane default.  As this goes through AT & RT we can see if that's a bad idea and it's low cost to switch back to required.

**Comment by dianabarsan** [comment]
I already added a comment about `DOCKER_CONFIG_PATH`: https://github.com/medic/cht-upgrade-service/pull/18#issuecomment-1281135898

> Is this for the [daemon config](https://docs.docker.com/config/daemon/#configure-the-docker-daemon) or the credentials to the AWS ECR config or something else?

yes!

**Comment by mrjones-plip** [comment]
Doing some more testing the `nginx` service can no longer talk to the `api` service because it's expecting it to be named `cht-api` but it's [now named just](https://github.com/medic/cht-core/blob/master/scripts/build/cht-core.yml.template#L37) `api`:

```
➜  4.0.0-alpha-compose docker logs cht_nginx_1
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh

[SNIP]

Launching Nginx                                                                                          │
/docker-entrypoint.sh: Configuration complete; ready for start up                                        │
2022/10/17 16:37:40 [emerg] 1#1: host not found in upstream "cht-api" in /etc/nginx/nginx.conf:57        │
nginx: [emerg] host not found in upstream "cht-api" in /etc/nginx/nginx.conf:57  
```

There's somehow a mismatch of compose [env var](https://github.com/medic/cht-core/blob/master/scripts/build/cht-core.yml.template#L81) `API_HOST` and the `$host` var [inside nginx's config](https://github.com/medic/cht-core/blob/master/nginx/nginx.conf#L20).

**Comment by dianabarsan** [comment]
Can you please update your docker-compose files with the new ones? New images were published today for `alpha`, which change nginx to take the api host as an env variable

**Comment by dianabarsan** [comment]
By the way, there's an e2e suite in `cht` that checks the interaction between CHT and the upgrade service: https://github.com/medic/cht-core/actions/runs/3265053363/jobs/5368283895

**Comment by mrjones-plip** [comment]
sweet! Thanks for the reminder to get latest images!  I nuked my image cache and re-ran the upgrade compose command - CHT started right up \o/!!

Cache has latest too of course:

```
➜  4.0.0-alpha-compose docker image ls
REPOSITORY                                        TAG           IMAGE ID       CREATED       SIZE
public.ecr.aws/s5s3h4s7/cht-nginx                 4.0.0-alpha   d5a9c2c54f32   5 hours ago   25.2MB
public.ecr.aws/s5s3h4s7/cht-haproxy-healthcheck   4.0.0-alpha   c87316573c02   5 hours ago   125MB
public.ecr.aws/s5s3h4s7/cht-haproxy               4.0.0-alpha   5d64f2bd612a   5 hours ago   499MB
public.ecr.aws/s5s3h4s7/cht-sentinel              4.0.0-alpha   bf67ef1c8573   5 hours ago   326MB
public.ecr.aws/s5s3h4s7/cht-api                   4.0.0-alpha   598b12987c48   5 hours ago   352MB
public.ecr.aws/s5s3h4s7/cht-upgrade-service       latest        e174b6b1981b   4 days ago    486MB
```

**Comment by mrjones-plip** [comment]
oh my - Sorry @dianabarsan  - you [caught the error further up](https://github.com/medic/cht-upgrade-service/pull/18#issuecomment-1281135898) and I didn't notice as the comments were flying by - thank you and sorry!

> I think you're missing DOCKER_CONFIG_PATH

File: README.md [hunk 1]
```diff
@@ -25,30 +25,33 @@ Save this file in a new folder on your computer - a different folder than the on
...
-| `COUCHDB_PASSWORD` | yes      | CouchDb main admin account password.                                                                                                                                                                                                       | 
+| Name            | Required | Description                                                                                                                                                                                                                                  |
+|-----------------|----------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `CHT_COMPOSE_PATH` | yes      | Absolute path to the folder where the CHT docker-compose file is saved.                                                                                                                                                                      |
```
[the remark this answers is not in the document]
Comment (mrjones-plip) [reply README.md:30]: Updated my suggestion to reflect this - thanks for the clarification!

**Review by mrjones-plip — APPROVED** [review]
lgtm! 

let's ship this out and I'll submit a PR to add the 4th `Example` column

**Comment by dianabarsan** [comment]
Awesome, thank you!

**Pull request merged**